### ISSUE TITLE
Import-Schnittstellen?

### ISSUE THREAD

**Issue by jfiala** [krabina/Datencockpit#20]
Sind von euch Import-Schnittstellen geplant?
Wie es schein wäre derzeit nur ein MediaWiki-Page-Import möglich.

Wir würden gerne Daten im Excel-Format importieren.

**Comment by krabina** [comment]
CSV-Import funktioniert über die Extension DataTransfer, die dabei ist:
https://www.mediawiki.org/wiki/Extension:Data_Transfer

Man muss in der CSV-Struktur den Vorlagen folgen und dien Namen der Vorlag in den Feldern angeben. Z. B. für eine Folgenabschätzung
Seitenname,Folgen[Datum],Folgen[betroffene Personenkreise],Folgen[Datenkategorien],...,Freitext

Wenn ich mal Zeit finde, könnte ich CSV-Vorlagen dafür erstellen....

**Comment by jfiala** [comment]
Wir haben unter https://github.com/knowyourdata/data-scanner ein Tool für die Ermittlung von sensitiven Tabellen/Spalten bereitgestellt. 

Unter https://github.com/knowyourdata/data-scanner/blob/master/data-scanner-core/xdocs/sample_northwind/result.xlsx
befindet sich ein Beispiel-Auswertungsergebnis mit der Nordwind-Datenbank von MS Access.

Bitte um Info wenn für das Datencockpit andere Spaltenbeschriftung bzw. eine andere Aufereitung notwendig wären. Die jetzige Darstellung auf Tabellenebene entspricht der Ebene "Daten" im Verfahrensverzeichnis des Datencockpit, zusätzlich hätten wir dann noch die Auswertung auf Spaltenebene.

**Comment by krabina** [comment]
Ich versteh das nicht ganz. Ein Data Scanner würde sensitive Daten in Excel-Tabellen erkenne. Ist toll, hat aber mit einem Import ins Datencockpit doch nichts zu tun. Im Datencockpit könnte man sinnvollerweise ein in Excel erstellte Verfahrensverzeichnis importieren, aber keine Daten mit sensitiven Daten...

**Comment by jfiala** [comment]
Nein, das Tool ermittelt nur die Tabellen/Spalten wo sensitive Daten gespeichert sind. Damit erspart man sich einerseits das händische Pflegen des Verarbeitungsverzeichnis bzw. entdeckt auch Stellen wo sonst noch sensitive Daten gespeichert sind. Also sozusagen wie das Datencockpit, aber bottom-up von der Datenbank weg. Damit würden sich die Tools gut ergänzen. Voraussetzung wäre, daß die Schnittstelle idealerweise auch additiv arbeiten kann, sprich nicht alles ersetzt sondern synchronisiert...

**Comment by krabina** [comment]
ah, verstehe! Da gäbe es wohl mehrere Wege:
Variante 1: händischer Import eines CSVs eures Tools durch den User. Dazu müsste man die oben beschriebene Struktur verwenden. Wichtig ist, UTF-16 zu verwenden, da der Import mit UTF-8 buggy ist bei Umlauten.
Variante 2: Das Datencockpit holt sich von eurem Tool die Daten und importiert sie automatisch. Dazu kann men die Extension [External Data](https://www.mediawiki.org/wiki/Extension:External_Data) verwenden, die im Datencockpit mit dabei ist.
Variante 3: Vermutlich der eleganteste Weg ist es, wenn euer Tool die API verwendet und direkt ins Datencockpit schriebt:  [MediaWiki-API](https://www.mediawiki.org/wiki/API:Main_page) bzw. https://www.mediawiki.org/wiki/API:Edit
